pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -222,12 +217,12 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 		generateEncryptionConfigurationMetaData = g.Add(flow.Task{
 			Name:         "Generating etcd encryption configuration",
 			Fn:           flow.TaskFn(botanist.GenerateEncryptionConfiguration).DoIf(enableEtcdEncryption),
-			Dependencies: flow.NewTaskIDs(deployNamespace, ensureShootStateExists, loadSecrets, generateSecrets),
+			Dependencies: flow.NewTaskIDs(deployNamespace, ensureShootStateExists),
```

Comment:
Why was `generateSecrets` removed?

Reply:
Don't need it for this step. `generateSecrets` is needed for the step which persists the ETCDEncrytionConfiguration as that step is the one reading/writing from/to the ShootState.
GenerateEncryptionConfiguration uses a variable to store only the generated configuration.

Reply:
OK, thanks!